Snowflake.Data/Client/SFCredentialManagerFactory.cs

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+namespace Snowflake.Data.Client
+{
+    using System;
+    using Snowflake.Data.Core;
+    using Snowflake.Data.Core.CredentialManager;
+    using Snowflake.Data.Core.CredentialManager.Infrastructure;
+    using Snowflake.Data.Log;
+    using System.Runtime.InteropServices;
+
+    public class SFCredentialManagerFactory
+    {
+        private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<SFCredentialManagerFactory>();
+
+        private static ISnowflakeCredentialManager s_customCredentialManager = null;
+
+        internal static string BuildCredentialKey(string host, string user, TokenType tokenType, string authenticator = null)
+        {
+            return $"{host.ToUpper()}:{user.ToUpper()}:{SFEnvironment.DriverName}:{tokenType.ToString().ToUpper()}:{authenticator?.ToUpper() ?? string.Empty}";
+        }
+
+        public static void UseDefaultCredentialManager()
+        {
+            s_logger.Info("Clearing the custom credential manager");
+            s_customCredentialManager = null;
+        }
+
+        public static void SetCredentialManager(ISnowflakeCredentialManager customCredentialManager)
+        {
+            s_logger.Info($"Setting the custom credential manager: {customCredentialManager.GetType().Name}");
```

Comment:
Done

Reply:
great. Maybe we could also have a check if the parameter is null?
I know that the error would be thrown anyway but maybe we could throw a nicer error of SnowflakeDbException instead of just null pointer exception.

Reply:
Added validation when customCredentialManager is set to null and used different message in that case.

Reply:
Fixed, changed implementation of credentialManagerFactory.